src/legacy/shared/GoogleMapsWithMarkers/GoogleMapsWithMarkers.tsx

```diff
@@ -25,12 +26,18 @@ interface State {
 }
 
 class GoogleMapsWithMarkers extends React.Component<Props, State> {
+  directionsService = new google.maps.DirectionsService();
+  directionsDisplay = new google.maps.DirectionsRenderer();
+
   state: State = {}
 
   handleMapMounted = (map: GoogleMap) => {
     if (!map) {
       return;
     }
+    const googleMap: google.maps.Map = map.context.__SECRET_MAP_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
```

Comment:
This is stupid. `react-google-maps` doesn't expose the `google.maps.Map`, but the `google.maps.DirectionsRenderer` needs it. `react-google-maps` is also unmaintained at this point. There are other React Google Maps libraries out there, but that makes the scope grow.

Here are the options I see:

1. Use the hack above (at the risk of *being fired*) and add an annotation to silence TypeScript's complaining.
2. Switch to another React Google Maps library (which is a slightly larger refactor which would block us here)
3. Use the Google Maps API directly from here (and elsewhere, progressively; think we started this with Autocomplete?)

The engineer in me is drawn toward Option 1, as it's the simplest progressive enhancement which does not preclude any other options.

The developer in me is drawn toward Option 2, because we're going to need to do that anyway and intermediary solutions are just overhead.

The programmer in me is drawn toward Option 3, because a React API for Google Maps doesn't do anything more than the JS API for Google Maps, so why bother?

So, soliciting advice!

Reply:
It seems all of our google maps code is abstracted behind the `GoogleMaps` and `GoogleAutoComplete` components so taking on `3` isn't a bad choice especially if the react api is just a wrapper and (unmaintained).

Reply:
firstly, i second, the third option also

Reply:
After all that, it turns out `react-google-maps` has a [`DirectionsRenderer` component](https://github.com/tomchentw/react-google-maps/blob/ca5c5c6f5346fb3ee0037893fbca488a408c9938/src/components/DirectionsRenderer.md), so there's not a need to try to get at the `google.maps.Map` directly.

Oops.

Reply:
lolol. react-google-maps has been so hassle as app grows ><.